core/commands/push.py

```diff
@@ -53,8 +53,21 @@ def run(self, force=False):
                     "set_upstream": True,
                     "force": force
                     })
+        elif self.get_upstream_for_active_branch():
+            upstream = self.get_upstream_for_active_branch()
+            current_local_branch = self.get_current_branch_name()
+            remote, branch = upstream.split("/", 1)
+            sublime.set_timeout_async(
+                lambda: self.do_push(remote, branch, local_branch=current_local_branch, force=force))
         else:
-            sublime.set_timeout_async(lambda: self.do_push(None, None, force=force))
+            # if `prompt_for_tracking_branch` is false, ask for a remote and perform
+            # push current branch to a remote branch with the same name
+            current_local_branch = self.get_current_branch_name()
```

Comment:
I know this has already been merged, but ideally this sort of operation would occur on the non-UI thread (`run_async` pattern).